api/dbadapters/bigquery.ts

```diff
@@ -91,22 +93,57 @@ export class BigQueryDbAdapter implements IDbAdapter {
       .promise();
   }
 
-  public async evaluate(statement: string) {
-    try {
-      await this.getClient().query({
-        useLegacySql: false,
-        query: statement,
-        dryRun: true
-      });
-      return dataform.QueryEvaluationResponse.create({
-        status: dataform.QueryEvaluationResponse.QueryEvaluationStatus.SUCCESS
-      });
-    } catch (e) {
-      return dataform.QueryEvaluationResponse.create({
-        status: dataform.QueryEvaluationResponse.QueryEvaluationStatus.FAILURE,
-        error: parseBigqueryEvalError(e)
-      });
+  public async evaluate(
+    query: string | dataform.Table | dataform.Operation | dataform.Assertion,
+    projectConfig?: dataform.IProjectConfig
+  ) {
+    let executionTasks: dataform.ExecutionTask[];
+    if (typeof query === "string") {
+      executionTasks = [dataform.ExecutionTask.create({ statement: query })];
+    } else {
+      try {
+        const coreAdapter = new BigQueryAdapter(projectConfig, version);
+        if (query instanceof dataform.Table) {
+          executionTasks = coreAdapter
+            .publishTasks(
+              query,
+              { useSingleQueryPerAction: projectConfig?.useSingleQueryPerAction },
+              null
+            )
+            .build();
+        } else if (query instanceof dataform.Operation) {
+          executionTasks = query.queries.map(statement =>
```

Comment:
OK, for this case i can see that we're going to need to return a list of evaluation responses, I'm fine with that API change!